lib/deploy/functions/deploy.js

```diff
@@ -35,6 +35,7 @@ module.exports = function(context, options, payload) {
         utils.logSuccess(chalk.green.bold('functions:') + ' ' + chalk.bold(options.config.get('functions.source')) + ' folder uploaded successfully');
       }).catch(function(err) {
         utils.logWarning(chalk.yellow('functions:') + ' Upload Error: ' + err.message);
```

Comment:
Do we need to log the warning at all anymore? I think this might have been around from a time when we wanted function deploys to fail without breaking the overall deploy, but I don't think that's the case anymore so we can probably just remove the catch block altogether, right?

Reply:
I think the error returned from uploading is rather unfriendly, so I'd prefer to still log warning so that users know that something went wrong with functions upload